app/models/user.rb

```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  scope :three_most_recent_posts_for_user, ->(user_id) { where(id: user_id).order(created_at: :desc).limit(3) }
+  has_many :posts
```

Comment:
as a requirement, the  author_id column in posts table should be the foreign_key for the users table😉.

<img width="944" alt="image" src="https://github.com/ismayito/Rails-blog-app/assets/64643995/748df7f6-77df-4701-972c-cbdc5638aaa6">



```suggestion
   has_many :posts, foreign_key: 'author_id'
```